llvm/lib/CodeGen/StackMaps.cpp

```diff
@@ -282,7 +284,26 @@ StackMaps::parseOperand(MachineInstr::const_mop_iterator MOI,
     const TargetRegisterClass *RC = TRI->getMinimalPhysRegClass(MOI->getReg());
     assert(!MOI->getSubReg() && "Physical subreg still around.");
 
-    unsigned Offset = 0;
+
+    signed Offset = 0;
+    // Check if there are any mappings for this register in the spillmap. If so,
+    // encode the additional location in the offset field of the stackmap record
+    // (which is unused for register locations). Note that this assumes that
+    // there can only be one additional location for each value, which may turn
+    // out to be false.
+    if (MOI->isReg()) {
+      Register R = MOI->getReg();
+      if (SpillOffsets.count(R) > 0) {
+        Offset = SpillOffsets[R];
+        if (Offset > 0) {
```

Comment:
Should we do something like `assert(SpillOffsets[R] == 0)` here, because if it's not-0, presumably our assumption that there's only one spill is wrong, and our code is broken?

Reply:
`SpillOffsets[R]` can be negative which means it's a stack spill, rather than a register spill. Previously it could also be `0` meaning there's no additional location, but I can't remember if that's still used.

Reply:
OK, then we might as well `assert(SpillOffsets[R] != 0)` and if `0` is meaningful, we'll soon find out :)

Reply:
Fixed in [9985278](https://github.com/ykjit/ykllvm/pull/57/commits/99852782f848b01f26aba142f0f6b2becf3b3e81)